src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -57,13 +57,34 @@ public static Process Start(string fileName, string arguments, string userName,
         /// <summary>Terminates the associated process immediately.</summary>
         public void Kill()
         {
-            EnsureState(State.HaveNonExitedId);
-            if (Interop.Sys.Kill(_processId, Interop.Sys.Signals.SIGKILL) != 0)
+            EnsureState(State.HaveId);
+
+            // Check if we know the process has exited. This avoids us targetting another
+            // process that has a recycled PID. This only checks our internal state, the Kill call below
+            // activly checks if the process is still alive.
+            if (GetHasExited(refresh: false))
+            {
+                return;
+            }
+
+            int killResult = Interop.Sys.Kill(_processId, Interop.Sys.Signals.SIGKILL);
+            if (killResult != 0)
             {
+                Interop.Error error = Interop.Sys.GetLastError();
+
+                // Don't throw if the process has exited.
+                if (error == Interop.Error.ESRCH)
+                {
+                    return;
+                }
+
```

Comment:
`GetLastError()` doesn't reset errno, does it? If it does, the line below should be `Win32Exception(error)`

Reply:
It doesn't change `errno`. This is an `Interop.Error` enum value. We can't pass this to `Win32Exception`. That one accepts the raw `int` errno value, from `Marshal.GetLastWin32Error`.